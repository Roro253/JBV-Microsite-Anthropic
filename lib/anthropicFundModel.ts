import { z } from "zod";

export const AnthFundSchema = z.object({
  company_meta: z.object({
    name: z.string(),
    mission: z.string(),
    tagline: z.string(),
    last_updated: z.string()
  }),
  public_kpis: z.object({
    valuation_post_money_usd: z.number().positive(),
    valuation_as_of: z.string(),
    arr_run_rate_usd: z.number().positive(),
    arr_as_of: z.string(),
    notes: z.string().optional()
  }),
  distribution_partners: z.array(
    z.object({
      name: z.string(),
      status: z.string(),
      link: z.string()
    })
  ),
  models: z.array(
    z.object({
      name: z.string(),
      as_of: z.string(),
      source: z.string()
    })
  ),
  safety: z.object({
    headline: z.string(),
    source: z.string()
  }),
  funding_context: z.object({
    series_f_usd: z.number().positive(),
    post_money_usd: z.number().positive(),
    as_of: z.string(),
    strategics: z.array(
      z.object({
        name: z.string(),
        amount_usd: z.number().nonnegative(),
        as_of: z.string().nullable(),
        source: z.string()
      })
    )
  }),
  investment_scenarios: z.object({
    defaults: z.object({
      commitment_usd: z.number().positive(),
      mgmt_fee_pct: z.number().min(0).max(1),
      carry_pct: z.number().min(0).max(1)
    }),
    entry: z.object({
      pre_or_post: z.string(),
      entry_valuation_usd: z.number().nullable(),
      ownership_pct: z.number().nullable(),
      note: z.string().optional()
    })
  }),
  market_cap_calculator: z
    .array(
      z.object({
        year: z.number().int(),
        revenue_run_rate_usd: z.number().positive()
      })
    )
    .min(1),
  market_comps: z.array(
    z.object({
      company: z.string(),
      note: z.string().optional()
    })
  ),
  sources: z.object({}).catchall(z.string())
});

export type AnthFundModel = z.infer<typeof AnthFundSchema>;

export async function getAnthropicFundModel(): Promise<AnthFundModel> {
  const rawModule = await import("@/data/anthropic_fund_model.json");
  const raw = (rawModule as { default?: unknown }).default ?? rawModule;
  const parsed = AnthFundSchema.safeParse(raw);
  if (!parsed.success) {
    console.error(parsed.error);
    throw new Error("Anthropic fund model JSON invalid");
  }
  return parsed.data;
}
