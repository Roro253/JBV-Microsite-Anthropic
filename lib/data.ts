import { cache } from "react";
import { z } from "zod";
import anthropicRaw from "@/data/anthropic.json";

const kpiMetricSchema = z.object({
  value: z.number(),
  as_of: z.string(),
  unit: z.string()
});

const anthropicSchema = z.object({
  company: z.object({
    name: z.string(),
    founded: z.string().optional(),
    structure: z.string(),
    mission: z.string(),
    tagline: z.string()
  }),
  leadership: z
    .array(
      z.object({
        name: z.string(),
        role: z.string(),
        bio: z.string()
      })
    )
    .min(1),
  product: z.object({
    platform: z.string(),
    highlights: z.array(z.string())
  }),
  kpis: z.object({
    run_rate_revenue: kpiMetricSchema,
    valuation: kpiMetricSchema,
    recent_round: z.object({
      round: z.string(),
      amount: z.number(),
      date: z.string(),
      lead: z.string()
    }),
    cloud_backers: z.array(z.string()),
    enterprise_adoption_notes: z.string()
  }),
  differentiators: z.array(z.string()),
  use_cases: z.array(z.string()),
  modes: z.object({
    explorer: z.object({ story: z.string() }),
    investor: z.object({ thesis: z.string() })
  }),
  spv_terms: z.object({
    min_check_usd: z.number(),
    fees: z.string(),
    closing_target: z.string(),
    docs_note: z.string()
  }),
  links: z.object({
    reserve_interest: z.string(),
    book_diligence: z.string()
  }),
  sources: z
    .array(
      z.object({
        claim: z.string(),
        url: z.string().url()
      })
    )
    .min(1),
  last_updated: z.string()
});

export type AnthropicData = z.infer<typeof anthropicSchema>;

let cachedData: AnthropicData | null = null;

const resolveLink = (value: string) => {
  if (value.startsWith("ENV:")) {
    const envKey = value.split("ENV:").at(1)?.trim();
    if (envKey) {
      const resolved = process.env[envKey];
      if (!resolved) {
        console.warn(
          `[anthropic-data] Missing environment variable ${envKey}. Did you update .env.local?`
        );
      }
      return resolved ?? "#";
    }
  }
  return value;
};

const fallbackData: AnthropicData = {
  company: {
    name: "Anthropic",
    founded: "2021",
    structure: "Public Benefit Corporation (PBC)",
    mission: "Build reliable, interpretable, and steerable AI systems that benefit humanity.",
    tagline: "Enterprise-grade, safety-first frontier AI."
  },
  leadership: [
    {
      name: "TBD",
      role: "Leadership placeholder",
      bio: "Content temporarily unavailable."
    }
  ],
  product: {
    platform: "Claude",
    highlights: []
  },
  kpis: {
    run_rate_revenue: { value: 0, as_of: "", unit: "USD" },
    valuation: { value: 0, as_of: "", unit: "USD" },
    recent_round: { round: "", amount: 0, date: "", lead: "" },
    cloud_backers: [],
    enterprise_adoption_notes: ""
  },
  differentiators: [],
  use_cases: [],
  modes: {
    explorer: { story: "" },
    investor: { thesis: "" }
  },
  spv_terms: {
    min_check_usd: 0,
    fees: "",
    closing_target: "",
    docs_note: ""
  },
  links: {
    reserve_interest: "#",
    book_diligence: "#"
  },
  sources: [],
  last_updated: new Date().toISOString().split("T")[0]
};

const hydrateData = () => {
  if (cachedData) {
    return cachedData;
  }

  try {
    const parsed = anthropicSchema.safeParse(anthropicRaw);
    if (!parsed.success) {
      console.warn("[anthropic-data] Validation failed", parsed.error.flatten());
      cachedData = fallbackData;
      return cachedData;
    }

    const data = parsed.data;
    cachedData = {
      ...data,
      links: {
        reserve_interest: resolveLink(data.links.reserve_interest),
        book_diligence: resolveLink(data.links.book_diligence)
      }
    };
    return cachedData;
  } catch (error) {
    console.warn("[anthropic-data] Failed to load data", error);
    cachedData = fallbackData;
    return cachedData;
  }
};

export const getAnthropicData = cache(hydrateData);
