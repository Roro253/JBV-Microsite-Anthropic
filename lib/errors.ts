export class ConfigurationError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "ConfigurationError";
  }
}

export function isConfigurationError(error: unknown): error is ConfigurationError {
  return error instanceof ConfigurationError;
}
