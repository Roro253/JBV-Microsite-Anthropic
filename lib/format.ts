export function formatUSDShort(n: number): string {
  const abs = Math.abs(n);
  if (abs >= 1e12) return `$${(n / 1e12).toFixed(1)}T`.replace(".0", "");
  if (abs >= 1e9) return `$${(n / 1e9).toFixed(1)}B`.replace(".0", "");
  if (abs >= 1e6) return `$${(n / 1e6).toFixed(1)}M`.replace(".0", "");
  return `$${n.toLocaleString()}`;
}

export function formatPct(p: number, opts: { sign?: boolean } = {}): string {
  const value = `${p.toFixed(0)}%`;
  if (opts.sign && p > 0) {
    return `+${value}`;
  }
  return value;
}

export function investedAfterFee(commitment: number, feePct: number): number {
  return commitment * (1 - feePct);
}

export function grossProceeds(invested: number, grossMoM: number): number {
  return invested * grossMoM;
}

export function jbvCarry(grossProfit: number, carryPct: number): number {
  return Math.max(0, grossProfit) * carryPct;
}

export function netToInvestors(
  commitment: number,
  feePct: number,
  grossMoM: number,
  carryPct: number
) {
  const invested = investedAfterFee(commitment, feePct);
  const gross = grossProceeds(invested, grossMoM);
  const grossProfit = gross - invested;
  const carry = jbvCarry(grossProfit, carryPct);
  const net = gross - carry;
  const netMoM = net / commitment;
  return { invested, gross, grossProfit, carry, net, netMoM };
}
