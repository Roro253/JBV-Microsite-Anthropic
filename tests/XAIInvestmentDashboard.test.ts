import "@testing-library/jest-dom/vitest";
import { beforeEach, describe, expect, test, vi } from "vitest";
import React, { type ReactNode } from "react";
import { render } from "@testing-library/react";

import XAIInvestmentDashboard from "@/components/XAIInvestmentDashboard";
import type { XFundModel } from "@/lib/xaiFundModel";
import { investedAfterFee, netToInvestors } from "@/lib/format";

vi.stubGlobal("React", React);
vi.stubGlobal(
  "ResizeObserver",
  class {
    observe() {}
    unobserve() {}
    disconnect() {}
  }
);

vi.mock("@/components/ui/tooltip", () => {
  const passthrough = ({ children }: { children?: ReactNode }) => React.createElement("div", null, children);
  return {
    TooltipProvider: passthrough,
    Tooltip: passthrough,
    TooltipTrigger: passthrough,
    TooltipContent: passthrough
  };
});

vi.mock("recharts", () => {
  const passthrough = ({ children }: { children?: ReactNode }) => React.createElement("div", null, children);
  return {
    ResponsiveContainer: passthrough,
    BarChart: passthrough,
    Bar: passthrough,
    CartesianGrid: () => null,
    Cell: () => null,
    LineChart: passthrough,
    Line: () => null,
    Tooltip: passthrough,
    XAxis: () => null,
    YAxis: () => null
  };
});

vi.mock("@/lib/hooks/use-reduced-motion", () => ({
  useReducedMotion: () => true
}));

const baseModel: XFundModel = {
  company_meta: {
    name: "xAI",
    mission: "",
    tagline: "",
    last_updated: "2025-10-12"
  },
  public_signals: {
    models: [],
    compute: {
      colossus_gpus_live: 150000,
      claim_built_days: 122,
      uptime_pct: 99,
      scale_outlook: "Expected to double in months",
      as_of: "2025-06-01"
    },
    fundraising_in_market: {
      amount_usd: 20000000000,
      mix: { equity_usd: 7500000000, debt_usd: 12500000000 },
      notable_participant: "Nvidia up to $2B",
      purpose: "Colossus 2 GPU procurement",
      status: "reported",
      as_of: "2025-10-07"
    },
    users_estimate: {
      maus_min: 30000000,
      maus_max: 39000000,
      as_of: "2025-09-30",
      note: "estimate"
    },
    world_model_plan: {
      target: "AI-generated video game by end-2026",
      hires: "Nvidia",
      as_of: "2025-10-12"
    },
    sources: {
      colossus: "https://x.ai/colossus"
    }
  },
  investment_scenarios: {
    defaults: {
      commitment_usd: 1_000_000,
      mgmt_fee_pct: 0.05,
      carry_pct: 0.1
    },
    entry: {
      pre_or_post: "scenario",
      entry_valuation_usd: null,
      ownership_pct: 10,
      note: ""
    }
  },
  market_cap_calculator: [
    { year: 2026, revenue_run_rate_usd: null },
    { year: 2027, revenue_run_rate_usd: null }
  ],
  market_comps: []
};

beforeEach(() => {
  localStorage.clear();
});

describe("xAI investment helpers", () => {
  test("invested after 5% fee", () => {
    expect(investedAfterFee(1_000_000, 0.05)).toBe(950_000);
  });

  test("ownership-based proceeds exceed invested capital", () => {
    const ownershipPct = 0.1;
    const marketCap = 200_000_000_000;
    const invested = investedAfterFee(1_000_000, 0.05);
    const gross = ownershipPct * marketCap;
    expect(gross).toBeGreaterThan(invested);
  });

  test("gross MoM fallback yields net MoM > 1", () => {
    const result = netToInvestors(1_000_000, 0.05, 3.0, 0.1);
    expect(result.netMoM).toBeGreaterThan(1);
  });
});

describe("xAI investment dashboard", () => {
  test("renders guided empty state when projections missing", () => {
    const { container, getByText } = render(
      React.createElement(XAIInvestmentDashboard, { fundModel: baseModel })
    );
    expect(getByText(/add revenue projections/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
