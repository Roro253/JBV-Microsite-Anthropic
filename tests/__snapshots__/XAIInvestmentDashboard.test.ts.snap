// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`xAI investment dashboard > renders guided empty state when projections missing 1`] = `
<div>
  <section
    class="space-y-6 rounded-3xl border border-sky-100 bg-gradient-to-b from-white via-white/90 to-sky-50/40 p-6 shadow-[0_40px_120px_-70px_rgba(15,23,42,0.4)]"
  >
    <header
      class="space-y-2 border-b border-slate-200 pb-4"
    >
      <p
        class="text-xs font-semibold uppercase tracking-[0.3em] text-sky-600"
      >
        Investor Mode · last updated 
        2025-10-12
      </p>
      <h2
        class="text-2xl font-semibold text-slate-900"
      >
        xAI Fund Model (Scenario)
      </h2>
      <p
        class="text-sm text-slate-600"
      >
        Public signals are sourced from xAI announcements and coverage. Scenario fields update as diligence progresses.
      </p>
    </header>
    <div
      class="rounded-3xl border border-sky-100 bg-white/80 p-6 shadow-[0_30px_80px_-60px_rgba(15,23,42,0.35)] backdrop-blur"
    >
      <div
        class="mb-4 space-y-1"
      >
        <h3
          class="text-lg font-semibold text-slate-900"
        >
          Public signals
        </h3>
        <p
          class="text-sm text-slate-600"
        >
          Facts reported by xAI or credible coverage.
        </p>
      </div>
      <div
        class="grid gap-3 md:grid-cols-2"
      >
        <div>
          <div>
            <button
              class="flex flex-col gap-1 rounded-2xl border border-sky-100 bg-white/70 p-4 text-left shadow-[0_20px_60px_-50px_rgba(15,23,42,0.25)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-300 transition hover:-translate-y-0.5"
              type="button"
            >
              <span
                class="inline-flex w-fit items-center rounded-full bg-sky-100 px-3 py-1 text-[11px] font-semibold uppercase tracking-[0.3em] text-sky-600"
              >
                official
              </span>
              <p
                class="text-sm font-semibold text-slate-900"
              >
                Compute
              </p>
              <p
                class="text-xs text-slate-600"
              >
                150,000 GPUs · 99% uptime · built in 122 days
              </p>
            </button>
          </div>
          <div>
            <p
              class="text-[11px] font-semibold uppercase tracking-[0.3em] text-slate-500"
            >
              Sources
            </p>
            <ul
              class="space-y-1"
            >
              <li>
                <a
                  class="text-sky-600 hover:text-sky-700"
                  href="https://x.ai/colossus"
                  rel="noopener"
                  target="_blank"
                >
                  xAI · Colossus (official site)
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="flex flex-col gap-1 rounded-2xl border border-sky-100 bg-white/70 p-4 text-left shadow-[0_20px_60px_-50px_rgba(15,23,42,0.25)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-300"
        >
          <span
            class="inline-flex w-fit items-center rounded-full bg-sky-100 px-3 py-1 text-[11px] font-semibold uppercase tracking-[0.3em] text-sky-600"
          >
            reported
          </span>
          <p
            class="text-sm font-semibold text-slate-900"
          >
            Fundraise
          </p>
          <p
            class="text-xs text-slate-600"
          >
            In market: up to $20.0B (equity ~$7.5B, debt ~$12.5B); Nvidia up to $2B (reported) · as of Oct 7, 2025
          </p>
        </div>
        <div
          class="flex flex-col gap-1 rounded-2xl border border-sky-100 bg-white/70 p-4 text-left shadow-[0_20px_60px_-50px_rgba(15,23,42,0.25)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-300"
        >
          <span
            class="inline-flex w-fit items-center rounded-full bg-sky-100 px-3 py-1 text-[11px] font-semibold uppercase tracking-[0.3em] text-sky-600"
          >
            official
          </span>
          <p
            class="text-sm font-semibold text-slate-900"
          >
            Models
          </p>
          <p
            class="text-xs text-slate-600"
          />
        </div>
        <div
          class="flex flex-col gap-1 rounded-2xl border border-sky-100 bg-white/70 p-4 text-left shadow-[0_20px_60px_-50px_rgba(15,23,42,0.25)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-300"
        >
          <span
            class="inline-flex w-fit items-center rounded-full bg-sky-100 px-3 py-1 text-[11px] font-semibold uppercase tracking-[0.3em] text-sky-600"
          >
            estimate
          </span>
          <p
            class="text-sm font-semibold text-slate-900"
          >
            Users
          </p>
          <p
            class="text-xs text-slate-600"
          >
            MAUs ~30–39M (estimates; not official) · as of Sep 30, 2025
          </p>
        </div>
        <div
          class="flex flex-col gap-1 rounded-2xl border border-sky-100 bg-white/70 p-4 text-left shadow-[0_20px_60px_-50px_rgba(15,23,42,0.25)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-300"
        >
          <span
            class="inline-flex w-fit items-center rounded-full bg-sky-100 px-3 py-1 text-[11px] font-semibold uppercase tracking-[0.3em] text-sky-600"
          >
            reported
          </span>
          <p
            class="text-sm font-semibold text-slate-900"
          >
            World models
          </p>
          <p
            class="text-xs text-slate-600"
          >
            Target: AI-generated video game by end-2026; Nvidia (reports)
          </p>
        </div>
      </div>
    </div>
    <div
      class="rounded-3xl border border-sky-100 bg-white/80 p-6 shadow-[0_30px_80px_-60px_rgba(15,23,42,0.35)] backdrop-blur"
    >
      <div
        class="mb-4 space-y-1"
      >
        <h3
          class="text-lg font-semibold text-slate-900"
        >
          Investment summary
        </h3>
        <p
          class="text-sm text-slate-600"
        >
          Scenario inputs; update when term sheet is finalized.
        </p>
      </div>
      <div
        class="grid gap-4 md:grid-cols-2"
      >
        <label
          class="flex flex-col gap-2 text-sm font-medium text-slate-700"
        >
          Entry valuation (USD)
          <input
            class="flex h-10 w-full rounded-md border border-sky-200 bg-white/80 px-3 py-2 text-sm text-slate-700 shadow-sm transition placeholder:text-slate-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:cursor-not-allowed disabled:opacity-50"
            inputmode="decimal"
            placeholder="Enter when available"
            type="text"
            value=""
          />
        </label>
        <label
          class="flex flex-col gap-2 text-sm font-medium text-slate-700"
        >
          Ownership %
          <input
            class="flex h-10 w-full rounded-md border border-sky-200 bg-white/80 px-3 py-2 text-sm text-slate-700 shadow-sm transition placeholder:text-slate-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:cursor-not-allowed disabled:opacity-50"
            inputmode="decimal"
            placeholder="Enter when available"
            type="text"
            value="10"
          />
        </label>
      </div>
      <p
        class="mt-3 text-xs text-slate-500"
      >
        When deal terms finalize, these fields auto-populate from 
        <code>
          xai_fund_model.json
        </code>
        .
      </p>
    </div>
    <div
      class="rounded-3xl border border-sky-100 bg-white/80 p-6 shadow-[0_30px_80px_-60px_rgba(15,23,42,0.35)] backdrop-blur"
    >
      <div
        class="mb-4 space-y-1 flex items-center justify-between"
      >
        <div>
          <h3
            class="text-lg font-semibold text-slate-900"
          >
            Market cap calculator
          </h3>
          <p
            class="text-sm text-slate-600"
          >
            Revenue is scenario-only; xAI does not disclose run-rate.
          </p>
        </div>
        <button
          class="inline-flex items-center gap-1 rounded-full border border-sky-200 px-3 py-1 text-xs font-semibold text-sky-600 hover:border-sky-300"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-pen-line h-3.5 w-3.5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 21h8"
            />
            <path
              d="M21.174 6.812a1 1 0 0 0-3.986-3.987L3.842 16.174a2 2 0 0 0-.5.83l-1.321 4.352a.5.5 0 0 0 .623.622l4.353-1.32a2 2 0 0 0 .83-.497z"
            />
          </svg>
          Finish editing
        </button>
      </div>
      <div
        class="grid gap-4 md:grid-cols-[minmax(0,280px)_1fr]"
      >
        <div
          class="space-y-4"
        >
          <label
            class="flex flex-col gap-2 text-sm font-medium text-slate-700"
          >
            Year
            <select
              class="rounded-xl border border-slate-200 bg-white px-3 py-2 text-sm text-slate-700 focus:border-sky-400 focus:outline-none focus:ring-2 focus:ring-sky-200"
            >
              <option
                value="2026"
              >
                2026
              </option>
              <option
                value="2027"
              >
                2027
              </option>
            </select>
          </label>
          <div
            class="space-y-2 text-sm font-medium text-slate-700"
          >
            <div
              class="flex items-center justify-between"
            >
              <span>
                EV/Sales multiple
              </span>
              <span>
                20
                ×
              </span>
            </div>
            <span
              aria-disabled="false"
              aria-label="EV to Sales multiple"
              class="relative flex w-full touch-none select-none items-center"
              data-orientation="horizontal"
              dir="ltr"
              style="--radix-slider-thumb-transform: translateX(-50%);"
            >
              <span
                class="relative h-1.5 w-full grow overflow-hidden rounded-full bg-sky-100"
                data-orientation="horizontal"
              >
                <span
                  class="absolute h-full bg-gradient-to-r from-sky-400 to-indigo-400"
                  data-orientation="horizontal"
                  style="left: 0%; right: 50%;"
                />
              </span>
              <span
                style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(50% + 0px);"
              >
                <span
                  aria-orientation="horizontal"
                  aria-valuemax="30"
                  aria-valuemin="10"
                  aria-valuenow="20"
                  class="block h-4 w-4 rounded-full border-2 border-white bg-sky-500 shadow-[0_4px_12px_-4px_rgba(32,118,199,0.6)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:pointer-events-none disabled:opacity-50"
                  data-orientation="horizontal"
                  data-radix-collection-item=""
                  role="slider"
                  style=""
                  tabindex="0"
                />
              </span>
            </span>
          </div>
          <label
            class="flex flex-col gap-2 text-sm font-medium text-slate-700"
          >
            Revenue run rate (USD)
            <input
              class="flex h-10 w-full rounded-md border border-sky-200 bg-white/80 px-3 py-2 text-sm text-slate-700 shadow-sm transition placeholder:text-slate-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:cursor-not-allowed disabled:opacity-50"
              inputmode="decimal"
              placeholder="Enter estimate"
              type="text"
              value=""
            />
          </label>
          <div
            aria-live="polite"
            class="rounded-2xl border border-sky-100 bg-white/70 p-4"
          >
            <p
              class="text-xs uppercase tracking-[0.3em] text-sky-600"
            >
              Implied market cap
            </p>
            <p
              class="mt-1 text-2xl font-semibold text-slate-900"
            >
              N/A
            </p>
            <p
              class="text-xs text-slate-500"
            >
              Scenario — enter a revenue estimate to compute market cap.
            </p>
          </div>
        </div>
        <div
          class="h-48"
        >
          <div>
            <div>
              <div />
              <div />
            </div>
          </div>
          <p
            class="mt-2 text-xs text-slate-500"
          >
            Provide a revenue estimate to illustrate potential market cap.
          </p>
        </div>
      </div>
    </div>
    <div
      class="rounded-3xl border border-sky-100 bg-white/80 p-6 shadow-[0_30px_80px_-60px_rgba(15,23,42,0.35)] backdrop-blur"
    >
      <div
        class="mb-4 space-y-1"
      >
        <h3
          class="text-lg font-semibold text-slate-900"
        >
          Waterfall returns
        </h3>
        <p
          class="text-sm text-slate-600"
        >
          Ownership-based when available; otherwise choose a scenario gross multiple.
        </p>
      </div>
      <div
        class="grid gap-6 lg:grid-cols-[minmax(0,260px)_1fr]"
      >
        <div
          class="space-y-4"
        >
          <label
            class="text-sm font-medium text-slate-700"
          >
            Commitment (USD)
            <input
              aria-label="Commitment in USD"
              class="flex h-10 w-full rounded-md border border-sky-200 bg-white/80 px-3 py-2 text-sm text-slate-700 shadow-sm transition placeholder:text-slate-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:cursor-not-allowed disabled:opacity-50"
              inputmode="decimal"
              type="text"
              value="1,000,000"
            />
          </label>
          <div
            class="space-y-2 text-sm font-medium text-slate-700"
          >
            <div
              class="flex items-center justify-between"
            >
              <span>
                Scenario gross MoM
              </span>
              <span>
                3.00
                ×
              </span>
            </div>
            <span
              aria-disabled="false"
              aria-label="Gross MoM multiplier"
              class="relative flex w-full touch-none select-none items-center"
              data-orientation="horizontal"
              dir="ltr"
              style="--radix-slider-thumb-transform: translateX(-50%);"
            >
              <span
                class="relative h-1.5 w-full grow overflow-hidden rounded-full bg-sky-100"
                data-orientation="horizontal"
              >
                <span
                  class="absolute h-full bg-gradient-to-r from-sky-400 to-indigo-400"
                  data-orientation="horizontal"
                  style="left: 0%; right: 77.77777777777777%;"
                />
              </span>
              <span
                style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(22.2222% + 0px);"
              >
                <span
                  aria-orientation="horizontal"
                  aria-valuemax="10"
                  aria-valuemin="1"
                  aria-valuenow="3"
                  class="block h-4 w-4 rounded-full border-2 border-white bg-sky-500 shadow-[0_4px_12px_-4px_rgba(32,118,199,0.6)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:pointer-events-none disabled:opacity-50"
                  data-orientation="horizontal"
                  data-radix-collection-item=""
                  role="slider"
                  style=""
                  tabindex="0"
                />
              </span>
            </span>
          </div>
          <div
            class="flex flex-wrap gap-2 text-xs text-slate-600"
          >
            <span
              class="inline-flex items-center rounded-full bg-sky-50 px-3 py-1 font-semibold text-sky-700"
            >
              Mgmt fee 
              5%
            </span>
            <span
              class="inline-flex items-center rounded-full bg-sky-50 px-3 py-1 font-semibold text-sky-700"
            >
              Carry 
              10%
            </span>
          </div>
        </div>
        <div
          class="space-y-4"
        >
          <div
            class="overflow-hidden rounded-2xl border border-sky-100"
          >
            <table
              class="min-w-full text-sm text-slate-700"
            >
              <tbody
                class="divide-y divide-slate-100"
              >
                <tr>
                  <td
                    class="px-4 py-2 text-slate-500"
                  >
                    Invested after fees
                  </td>
                  <td
                    class="px-4 py-2 text-right font-semibold text-slate-800"
                  >
                    $950,000
                  </td>
                </tr>
                <tr>
                  <td
                    class="px-4 py-2 text-slate-500"
                  >
                    Total gross proceeds
                  </td>
                  <td
                    class="px-4 py-2 text-right font-semibold text-slate-800"
                  >
                    $2.9M
                  </td>
                </tr>
                <tr>
                  <td
                    class="px-4 py-2 text-slate-500"
                  >
                    Gross profit
                  </td>
                  <td
                    class="px-4 py-2 text-right font-semibold text-slate-800"
                  >
                    $1.9M
                  </td>
                </tr>
                <tr>
                  <td
                    class="px-4 py-2 text-slate-500"
                  >
                    JBV carry
                  </td>
                  <td
                    class="px-4 py-2 text-right font-semibold text-slate-800"
                  >
                    $190,000
                  </td>
                </tr>
                <tr>
                  <td
                    class="px-4 py-2 text-slate-500"
                  >
                    Net to investors
                  </td>
                  <td
                    class="px-4 py-2 text-right font-semibold text-slate-800"
                  >
                    $2.7M
                  </td>
                </tr>
                <tr>
                  <td
                    class="px-4 py-2 text-slate-500"
                  >
                    Net MoM
                  </td>
                  <td
                    class="px-4 py-2 text-right font-semibold text-slate-800"
                  >
                    2.66×
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="h-36"
          >
            <div>
              <div>
                <div />
                <div />
                <div />
                <div />
              </div>
            </div>
          </div>
          <p
            class="text-xs text-slate-500"
          >
            Ownership path uses market cap × ownership %. Otherwise, MoM slider drives scenario returns.
          </p>
        </div>
      </div>
    </div>
    <div
      class="rounded-3xl border border-sky-100 bg-white/80 p-6 shadow-[0_30px_80px_-60px_rgba(15,23,42,0.35)] backdrop-blur"
    >
      <div
        class="mb-4 space-y-1 flex items-center justify-between"
      >
        <div>
          <h3
            class="text-lg font-semibold text-slate-900"
          >
            xAI projections
          </h3>
          <p
            class="text-sm text-slate-600"
          >
            Enter or toggle revenue projections to visualize growth scenarios.
          </p>
        </div>
        <div
          class="inline-flex rounded-full border border-sky-200 bg-white p-1"
        >
          <button
            class="rounded-full px-3 py-1 text-xs font-semibold transition bg-sky-600 text-white shadow"
            type="button"
          >
            Absolute
          </button>
          <button
            class="rounded-full px-3 py-1 text-xs font-semibold transition text-slate-500 hover:text-sky-600"
            type="button"
          >
            YoY %
          </button>
        </div>
      </div>
      <div
        class="space-y-3 rounded-2xl border border-sky-100 bg-white/70 p-4 text-sm text-slate-600"
      >
        <p
          class="font-semibold text-slate-700"
        >
          Add revenue projections
        </p>
        <p
          class="text-xs text-slate-500"
        >
          xAI has not disclosed revenue. Enter scenario run-rate estimates below to unlock charts.
        </p>
        <div
          class="grid gap-3 md:grid-cols-3"
        >
          <label
            class="text-xs font-medium text-slate-600"
          >
            2026
            <input
              class="flex h-10 w-full rounded-md border border-sky-200 bg-white/80 px-3 py-2 text-sm text-slate-700 shadow-sm transition placeholder:text-slate-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:cursor-not-allowed disabled:opacity-50 mt-1"
              inputmode="decimal"
              placeholder="USD"
              type="text"
              value=""
            />
          </label>
          <label
            class="text-xs font-medium text-slate-600"
          >
            2027
            <input
              class="flex h-10 w-full rounded-md border border-sky-200 bg-white/80 px-3 py-2 text-sm text-slate-700 shadow-sm transition placeholder:text-slate-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-sky-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white disabled:cursor-not-allowed disabled:opacity-50 mt-1"
              inputmode="decimal"
              placeholder="USD"
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
    </div>
    <div
      class="rounded-3xl border border-sky-100 bg-white/80 p-6 shadow-[0_30px_80px_-60px_rgba(15,23,42,0.35)] backdrop-blur"
    >
      <div
        class="mb-4 space-y-1"
      >
        <h3
          class="text-lg font-semibold text-slate-900"
        >
          Peer comparables
        </h3>
        <p
          class="text-sm text-slate-600"
        >
          Contextual peers; click through to available microsites.
        </p>
      </div>
      <div
        class="grid gap-4 md:grid-cols-3"
      />
    </div>
    <footer
      class="space-y-2 rounded-2xl border border-slate-200 bg-white/85 p-4 text-xs text-slate-600"
    >
      <p
        class="font-semibold text-slate-700"
      >
        Sources
      </p>
      <ol
        class="list-decimal space-y-1 pl-5"
      >
        <li>
          <span
            class="font-semibold uppercase tracking-wide text-slate-500"
          >
            colossus
            : 
          </span>
          <a
            class="text-sky-600 hover:text-sky-700"
            href="https://x.ai/colossus"
            rel="noopener"
            target="_blank"
          >
            https://x.ai/colossus
          </a>
        </li>
      </ol>
    </footer>
  </section>
</div>
`;
